src/main/java/com/atlassian/db/replica/api/state/States.java

```diff
@@ -0,0 +1,8 @@
+package com.atlassian.db.replica.api.state;
+
+public enum States {
```

Comment:
I probably should not use enums for API.

Reply:
Hmm, but transitions would be tricky: `getReadConnection()` would have to return both a `Connection` and a new `ConnectionState`. Or become mutable.

Reply:
What do you mean? Can you express it with a test case?